src/lib/UniversalDiscovery/Event/Subscriber/SectionAssign.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace EzSystems\EzPlatformAdminUi\UniversalDiscovery\Event\Subscriber;
+
+use eZ\Publish\API\Repository\ContentTypeService;
+use eZ\Publish\API\Repository\Exceptions\NotFoundException;
+use eZ\Publish\API\Repository\PermissionResolver;
+use eZ\Publish\API\Repository\Values\User\Limitation\ContentTypeLimitation;
+use EzSystems\EzPlatformAdminUi\UniversalDiscovery\Event\ConfigResolveEvent;
+use EzSystems\EzPlatformAdminUi\Util\PermissionUtil;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+class SectionAssign implements EventSubscriberInterface
+{
+    /** @var array */
+    private $allowedContentTypesIdentifiers;
+
+    /** @var \EzSystems\EzPlatformAdminUi\Util\PermissionUtil */
+    private $permissionUtil;
+
+    /** @var \eZ\Publish\API\Repository\ContentTypeService */
+    private $contentTypeService;
+
+    /**
+     * @param \eZ\Publish\API\Repository\PermissionResolver $permissionResolver
+     * @param \EzSystems\EzPlatformAdminUi\Util\PermissionUtil $permissionUtil
+     * @param \eZ\Publish\API\Repository\ContentTypeService $contentTypeService
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException
+     */
+    public function __construct(
+        PermissionResolver $permissionResolver,
+        PermissionUtil $permissionUtil,
+        ContentTypeService $contentTypeService
+    ) {
+        $this->permissionUtil = $permissionUtil;
+        $this->contentTypeService = $contentTypeService;
+        $hasAccess = $permissionResolver->hasAccess('section', 'assign');
+        $this->allowedContentTypesIdentifiers = $this->checkAllowedContentTypes($hasAccess);
+    }
+
+    /**
+     * @return array
+     */
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            ConfigResolveEvent::NAME => ['onUdwConfigResolve'],
+        ];
+    }
+
+    /**
+     * @param \EzSystems\EzPlatformAdminUi\UniversalDiscovery\Event\ConfigResolveEvent $event
+     */
+    public function onUdwConfigResolve(ConfigResolveEvent $event): void
+    {
+        $configName = $event->getConfigName();
+        if ('single' !== $configName && 'multiple' !== $configName) {
```

Comment:
I'm not sure if we need to check for `single` - I think we have only `multiple`.